scripts/python/phdf.py

```diff
@@ -110,24 +110,33 @@ def __init__(self, filename):
 
             # Read in coordinates
             tmpx = f['/Locations/x'][:,:]
-            self.x = np.zeros((self.NumBlocks,self.MeshBlockSize[0]))
-            for bId in range(self.NumBlocks):
-                for cId in range(self.MeshBlockSize[0]):
-                    self.x[bId,cId] = 0.5*(tmpx[bId,cId]+tmpx[bId,cId+1])
+            if '/VolumeLocations/x' in f:
+                self.x = f['/VolumeLocations/x'][:,:]
+            else:
+                self.x = np.zeros((self.NumBlocks,self.MeshBlockSize[0]))
+                for bId in range(self.NumBlocks):
+                    for cId in range(self.MeshBlockSize[0]):
+                        self.x[bId,cId] = 0.5*(tmpx[bId,cId]+tmpx[bId,cId+1])
```

Comment:
I think you could put this logic into a function and pass in an index, and simply return the the numpy array. This would reduce the repetition below.